app/core/tests/test_api.py

```diff
@@ -381,3 +382,14 @@ def test_create_soc_major(auth_client):
     res = auth_client.post(SOC_MAJOR_URL, payload)
     assert res.status_code == status.HTTP_201_CREATED
     assert res.data["title"] == payload["title"]
+
+
+def test_project_sdg_xref(auth_client, project, sdg):
+    project.sdgs.add(sdg)
+    project.save()
+
+    proj_res = auth_client.get(PROJECT_URL)
+    sdg_res = auth_client.get(SDG_URL)
+
+    assert filter(lambda proj: str(proj["uuid"]) == str(project.pk), proj_res.data)
```

Comment:
It looks like this is always going to pass since it's checking if the projects list contains the project model's pk.

I think you might have wanted to check if the projects list has a project that matches the sdg.projects models or one of them. I haven't given it much though on how to best test this.

This seems to work from my trial and error attempts:
```
    assert proj_res.data[0]["sdgs"][0] == sdg.name
```

### Possible missing API functionality

1. I was thinking "How would an API client assign an sdg to a project?" Do that API call and then check in the database to make sure it's really assigned. This might reveal that you need more code (in the serializer or somewhere) to help make that assignment.

2. Also, "How can an API client get the projects related to an sdg?" I found the `StringRelatedField` method as well, which returns the related project names. Maybe that's enough. It feels like we need something more than the name of the related sdg, like the pk instead, or as well. Suppose someone's doing an audit and wants `all the projects that are working on a specific sdg (sustainability goal)`. Maybe they want the project urls in addition to the names.

What do you think of these? I'm pretty sure the first one is a must, even though we don't have the explicit requirement to have it.

The generic thing to do is to return the pks of all the related objects so that a client can go look them up if they want the details like project urls.

We should bring this up at the meeting and push for some API design. With Django, we get the single table CRUD actions for cheap, but related data requires code for both assignment and retrieval in each direction. We don't need to implement them if they're not going to end up being called by a client.

Reply:
Thanks for catching this. Going forward, I'll test both cases, with and without the code changes to prevent this. 

As for the 2 points you made, I agree that 1 is absolutely required and 2 has to be discussed more. It was something that came up while I was working on it, but it's not built into the docs or mandated in the task. 

More specifically on point 2, I can imagine a case where the user may want to view an sdg and related projects on a single page. In that case you likely want more information than just the name, especially considering the project has many other fields. Although, I have no clue what the end user is doing as I haven't been to those meetings. I'm just implementing the models and relationships 😄.

It might take some research to see exactly how to implement more complex CRUD like this and document that, possibly a task itself. Then, creating tasks to actually implement it.

Reply:
Going back over what I did originally with the filter function: I did ```filter(lambda proj: str(proj["uuid"]) == str(project.pk), proj_res.data)``` to make sure the test case doesn't assume I know how many projects are in the database at the time. The current change clearly still needs work, but it would make it more robust to changes if and when someone adds more projects in the pytest fixture. Let me know what you think. 

I plan to implement the change suggested here:
> 
> ```
>     assert proj_res.data[0]["sdgs"][0] == sdg.name
> ```

but this is after I find the correct sdg in the ```proj_res```. Maybe I'm overdoing or overthinking it, but I'm willing to work together on something that works. I'll push new changes so you can see what I'm thinking as a whole.